controllers/controllers/workloads/cforg_controller.go

```diff
@@ -0,0 +1,172 @@
+	Log    logr.Logger
+}
+
+const (
+	StatusConditionReady  = "Ready"
+	OrgNameLabel          = "cloudfoundry.org/org-name"
+	hierarchyMetadataName = "hierarchy"
+	APIVersion            = "workloads.cloudfoundry.org/v1alpha1"
+	Kind                  = "CFOrg"
+)
+
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=subnamespaceanchors,verbs=list;create;delete;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=hierarchyconfigurations,verbs=get;list;watch;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the CFOrg object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *CFOrgReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	cfOrg := new(workloadsv1alpha1.CFOrg)
+	err := r.Client.Get(ctx, req.NamespacedName, cfOrg)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Error when trying to fetch CFOrg %s/%s", req.Namespace, req.Name))
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if cfOrg.ObjectMeta.DeletionTimestamp != nil && !cfOrg.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	var anchor v1alpha2.SubnamespaceAnchor
+	err = r.Client.Get(ctx, req.NamespacedName, &anchor)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			r.Log.Error(err, fmt.Sprintf("Error getting SubnamespaceAnchor for CFOrg %s/%s", req.Namespace, req.Name))
+			return ctrl.Result{}, err
+		}
+
+		anchor = v1alpha2.SubnamespaceAnchor{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      req.Name,
+				Namespace: req.Namespace,
+				Labels: map[string]string{
+					OrgNameLabel: cfOrg.Spec.Name,
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: APIVersion,
+						Kind:       Kind,
+						Name:       cfOrg.Name,
+						UID:        cfOrg.GetUID(),
+					},
+				},
+			},
+		}
+
+		err = r.Client.Create(ctx, &anchor)
+		if err != nil {
+			r.Log.Error(err, fmt.Sprintf("Error creating SubnamespaceAnchor for CFOrg %s/%s", req.Namespace, req.Name))
+			return ctrl.Result{}, err
+		}
+
+		setStatusConditionOnLocalCopy(&cfOrg.Status.Conditions, StatusConditionReady, metav1.ConditionUnknown, StatusConditionReady, "")
+		err = r.Client.Status().Update(ctx, cfOrg)
```

Comment:
Would using `r.Client.Status().Patch()` reduce potential update conflicts?

Reply:
Do you mean conflicting with the update status called at the end of this function? Actually it will never hit that after creating the subnamespace anchor because it isn't being requeried (And so its Status.State param cannot be OK). So I'm actually going to return and requeue after creating the subnamespace anchor so its clear it should check its status on the next Reconcile loop.